File: Easy/JavaScript/2319. Check if Matrix Is X-Matrix.js
var checkXMatrix = function(grid) {
    let x = true;
    let n = grid.length;

    for(let i = 0; i < n; i++){
        for(let j = 0; j < n; j++){
            if((i == j || (i + j) == n - 1) && grid[i][j] == 0){
                x = false;
            }else if(i != j && (i + j) != n - 1 && grid[i][j] != 0){
                x = false;
            }
        }
    }

    return x;
};

File: Easy/Python/2319. Check if Matrix Is X-Matrix.py
class Solution(object):
    def checkXMatrix(self, grid):
        x = True
        n = len(grid)

        for i in range(n):
            for j in range(n):
                if (i == j or (i + j) == n - 1) and grid[i][j] == 0:
                    x = False
                elif i != j and (i + j) != (n - 1) and grid[i][j] != 0:
                    x = False

        return x
